master/internal/api_notebook.go

```diff
@@ -27,3 +36,49 @@ func (a *apiServer) KillNotebook(
 	_ context.Context, req *apiv1.KillNotebookRequest) (resp *apiv1.KillNotebookResponse, err error) {
 	return resp, a.actorRequest(fmt.Sprintf("/notebooks/%s", req.NotebookId), req, &resp)
 }
+
+func (a *apiServer) NotebookLogs(
+	req *apiv1.NotebookLogsRequest, resp apiv1.Determined_NotebookLogsServer) error {
+	if err := grpc.ValidateRequest(
+		grpc.ValidateLimit(req.Limit),
+	); err != nil {
+		return err
+	}
+
+	cmdManagerAddr := actor.Addr("notebooks", req.NotebookId)
+	eventManager := a.m.system.Get(cmdManagerAddr.Child("events"))
+
+	logRequest := api.LogsRequest{
+		Offset: int(req.Offset),
+		Limit:  int(req.Limit),
+		Follow: req.Follow,
+	}
+
+	onLogEntry := func(log *logger.Entry) error {
+		return resp.Send(&apiv1.NotebookLogsResponse{LogEntry: api.LogEntryToProtoLogEntry(log)})
+	}
+
+	streamID, err := uuid.NewUUID()
+	if err != nil {
+		return status.Errorf(codes.Internal, "failed to generate the stream uuid")
+	}
+	logStreamActorAddr := cmdManagerAddr.Child("logStream-" + streamID.String())
+	logStreamActor, created := a.m.system.ActorOf(
+		logStreamActorAddr,
+		api.NewLogStreamActor(
+			resp.Context(),
+			eventManager,
+			logRequest,
+			onLogEntry,
+		),
+	)
+
+	if !created {
+		return errors.New("failed to create actor")
+	}
+
+	// We delegate checking for context closure to logStreamActor accepting that
+	// it could stay up if there are no log messages coming in instead of busy checking
+	// to kill the actor.
```

Comment:
I don't think this comment is really necessary.  If you'd like to keep it, let's shorten it to "keep the GRPC context open until the logStream actor stops" or something?